lib/remap-functions/src/string.rs

```diff
@@ -0,0 +1,45 @@
+use remap::prelude::*;
+
+#[derive(Clone, Copy, Debug)]
+pub struct STring;
```

Comment:
This is a new function (named `STring` to not clash with `std::String` 👅) that's part of our story to enhance type safety.

Basically, because functions now expect arguments to have the correct type, you need a way to guarantee such a type.

We already have `to_string(.foo)`, but that's a _coercion_ function, whereas the new `string(.foo)` is a _type checking_ functions.

So for example, instead of doing this:

```rust
parse_json(.foo)
```

You could do this:

```rust
parse_json(to_string(.foo) ?? "{}")
```

But that still leaves the possibility of any non-string being coerced into a string, which might not be what you expect.

So now you can do this:

```rust
parse_json(string(.foo) ?? "{}")
```

Which guarantees that `.foo` is only parsed as JSON if it is actually a string and no other type.

Note that like all functions, you still need to store the resulting value for the type check to be tracked.

So, this won't work:

```rust
string!(.foo)
parse_json(.foo) // type error: .foo is of unknown type
```

But this will:

```rust
.foo = string!(.foo)
parse_json(.foo)
```

The reason for this is that `string` is a regular function like any other, and we don't want functions to mutate the type state of the program, as that can lead to bad results (e.g. it would invalidate all our type safety guarantees) if a function is badly implemented.

Instead, the resulting value is captured by the assignment expression, and _it_ handles the type definition of `.foo` for the rest of the program.

@binarylogic and I discussed providing a built-in solution to type checking, but we decided against that for now as we like the simplicity of everything being done through functions for now, and we also don't have a nice syntactic solution yet to add type information into the language. We might still do this in the future, but not for now.